src/Microsoft.AspNet.Http.Interfaces/IRequestIdentifierFeature.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Http.Interfaces
+{
+    /// <summary>
+    /// Feature to identify a request.
+    /// </summary>
+    public interface IRequestIdentifierFeature
+    {
+        /// <summary>
+        /// Identifier to trace a request.
+        /// </summary>
+        Guid TraceIdentifier { get; }
```

Comment:
The nearly 100% use case is that they're persisted in an ETW log or similar structure, so they're definitely not persisted as strings. Please reread my argument above - I'm not advocating GUIDs for the sake of having GUIDs, I'm advocating it because it's structured, unambiguous, and easy for correlation engines to reason about. If you have another structure in mind that meets these requirements then by all means suggest it.